clang/include/clang/3C/ProgramInfo.h

```diff
@@ -113,12 +113,25 @@ class ProgramInfo : public ProgramVariableAdder {
   void constrainWildIfMacro(ConstraintVariable *CV, SourceLocation Location,
                             PersistentSourceLoc *PSL = nullptr);
 
+
+  void unifyIfTypedef(const clang::Type*, clang::ASTContext&,
+                      clang::DeclaratorDecl*, PVConstraint*);
+
+  std::pair<CVarSet, bool> lookupTypedef(PersistentSourceLoc PSL);
+
+  bool seenTypedef(PersistentSourceLoc PSL);
+
+  void addTypedef(PersistentSourceLoc PSL, bool ShouldCheck);
+
 private:
   // List of constraint variables for declarations, indexed by their location in
   // the source. This information persists across invocations of the constraint
   // analysis from compilation unit to compilation unit.
   VariableMap Variables;
 
+  // Map storing constraint information for typedefed types
+  std::map<PersistentSourceLoc, std::pair<CVarSet, bool>> typedefVars;
```

Comment:
What is this "constraint information?" Can you explain what the `CVarSet` and `bool` are used for? I presume that the `CVarSet` is one or more constraint variables that represent the types being defined e.g., representing `int *` in `typedef int *ptr_int_t;` What's the `bool` for? Why do you need a set of constraint variables?